devcheck

```diff
@@ -15,9 +16,29 @@
 
 # This script runs all tests and examples and verifies they work as expected.
 
-[[ "${BASH_SOURCE[0]}" != "${0}" ]] && EXIT=return || EXIT=exit
+python3 -m unittest discover --start-directory . --pattern '*_test.py' --verbose
+
+if [ $? -ne 0 ]; then
+  echo "Test failed with code $?"
+  exit
+fi
 
-python3 -m unittest discover -s . -p '*_test.py' -v || { RESULT=$? ; echo -e "\n\n Test failed with code $RESULT\n\n" ; $EXIT $RESULT ; }
+tests_passed=true
 
- eval $(find examples/ -name 'run.sh' -printf 'echo -e "\\n\\n+++++++++++ Running: %p" && %p && ') echo -e '+++++\\n\\nChecks: OK' || { RESULT=$? ; echo -e "+++++\n\nChecks: ERROR (status: $RESULT) above" ; $EXIT $RESULT ; }
+for test_script in `find examples/ -type f -name run.sh`; do
+  test_directory="$( dirname "$test_script" )"
+  example_name="$( basename "$test_directory" )"
+  echo "Running $example_name"
```

Comment:
nit: Having some sort of visual demarcation (eg `+++++++++++`) might be useful to see the example boundaries.

Reply:
Used colors for easy to see boundaries 🎨

Please let me know if terminal colors are not an effective visual cue for you. I'll add separators if you want! But take a look at the screenshot first.

Reply:
I like the colors! Let's keep them in there, but please add the separators as well for color-blind folks. I want the tool to be widely accessible.